packages/filebrowser/src/model.ts

```diff
@@ -103,8 +98,15 @@ export class FileBrowserModel implements IDisposable {
       }
     };
     window.addEventListener('beforeunload', this._unloadEventListener);
-    this._scheduleUpdate();
-    this._startTimer();
+    this._poll = new Poll({
+      factory: () => this.cd('.'),
```

Comment:
I think the problem is really that this factory function should be async. It's not waiting for the cd to finish.

Reply:
No, that was dumb. It is returning a promise, so that should be good.

Reply:
It is async, since `this.cd('.')` returns a promise. In some of my testing it was explicitly marked as `async`, which made no difference.